datafusion/sqllogictest/test_files/update.slt

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+##########
+## Update Tests
+##########
+
+statement ok
+create table t1(a int, b varchar, c double);
+
+# Turn off the optimizer to make the logical plan closer to the initial one
+statement ok
+set datafusion.optimizer.max_passes = 0;
+
+query TT
+explain update t1 set a=1, b=2, c=3.0;
```

Comment:
```suggestion
explain update t1 set a=1, b=2, c=3.0, d=NULL;
```

Reply:
Could we add a test for NULL 🤔?

Reply:
Thank you @Weijun-H. I think it would be helpful.

Reply:
NULL will also be cast, but this casting will be optimized later.
```sh
❯ explain update t1 set a=1, b=2, c=3.0, d=NULL;
+--------------+--------------------------------------------------------------------------------+
| plan_type    | plan                                                                           |
+--------------+--------------------------------------------------------------------------------+
| logical_plan | Dml: op=[Update] table=[t1]                                                    |
|              |   Projection: Int32(1) AS a, Utf8("2") AS b, Float64(3) AS c, Int32(NULL) AS d |
|              |     TableScan: t1 projection=[]                                                |
+--------------+--------------------------------------------------------------------------------+
1 row in set. Query took 0.034 seconds
```